crates/sui-indexer-alt-graphql/src/api/types/available_range.rs

```diff
@@ -46,30 +44,387 @@ impl AvailableRange {
+        $(|$pipes:ident, $filt:ident| $block:block)?
+        ;
+    )*) => {
+        /// Populates `pipelines` with indexer pipeline names by matching against the collect_pipelines! macro configuration.
+        fn collect_pipelines(
+            type_: &str,
+            field: Option<&str>,
+            filters: &mut BTreeSet<String>,
+            pipelines: &mut BTreeSet<String>,
+        ) {
+            match (type_, field) {
+                $(
+                    (stringify!($type), _field @ Some($(stringify!($field)) |*)) => {
+                        $(
+                            $($(
+                                filters.insert($f_to_add.to_string());
+                            )?)?
+                            let delegate_field = if stringify!($delegate_field) == "*" {
+                                _field
+                            } else {
+                                Some(stringify!($delegate_field))
+                            };
+                            collect_pipelines(stringify!($delegate_type), delegate_field, filters, pipelines);
+                        )?
+                        $(
+                            let $filt = filters;
+                            let $pipes = pipelines;
+                            $block
+                        )?
+                    }
+                )*
+                _ => {}
+            }
+        }
+
+        #[cfg(test)]
+        mod field_piplines_tests {
+            use crate::schema;
+            use async_graphql::{
+                extensions::{Extension, ExtensionContext, ExtensionFactory, NextRequest},
+                registry::{MetaType, Registry},
+                Response,
+            };
+            use std::{collections::BTreeSet, sync::Arc};
+
+            // Collect macro expansion into a vector of tuples for testing.
+            const TYPE_FIELD_DELEGATIONS: &[(&str, &[&str], Option<(&str, &str)>)] = &[
+            $(
+                (
+                    stringify!($type),
+                    &[$(stringify!($field)),*],
+                    {
+                        #[allow(unused_assignments)]
+                        let _delegation: Option<(&str, &str)> = None;
+                        $(
+                            let _delegation = Some((stringify!($delegate_type), stringify!($delegate_field)));
+                        )?
+                        _delegation
+                    }
+                ),
+                )*
+            ];
+
+            /// Validates that all types, fields in the macro exist in the GraphQL schema
+            /// and outputs a snapshot of all type.field (filter) -> pipeline mappings for regression testing.
+            ///
+            /// Test to ensure the macro configuration stays in sync with the GraphQL schema. If a type, field, or
+            /// filter is referenced in the macro but doesn't exist in the schema, this test will fail.
```

Comment:
nit: The formatting here seems a bit strange -- like these are two versions of the same doc comment.